models/demos/falcon7b/tt/model_config.py

```diff
@@ -129,6 +130,55 @@ def get_model_config(model_config_str, prefill_seq_len=0):
     model_config.update({f"{key}_MEMCFG": mem_config for key in OP_KEYS if key not in NO_MEMCFG})
     model_config.update({f"{key}_DTYPE": dtype for key in OP_KEYS if key not in NO_DTYPE})
 
+    ln_max_num_cores_y = 8
+    for i in range(ln_max_num_cores_y, 0, -1):
+        if (prefill_seq_len // 32) % i == 0:
+            ln_num_cores_y = i
+            break
+
+    ln_num_cores_x = 8
+    hidden_dim = 4544
+
+    num_tiles_per_core_h = math.ceil(prefill_seq_len / ln_num_cores_y / 32)
+    num_tiles_per_core_w = math.ceil(hidden_dim / ln_num_cores_x / 32)
+    ln_shard_height_hidden_dim = num_tiles_per_core_h * 32
+    ln_shard_width_hidden_dim = num_tiles_per_core_w * 32
+
+    core_range_block_sharded_layernorm = ttnn.experimental.tensor.CoreRangeSet(
+        {
+            ttnn.experimental.tensor.CoreRange(
+                ttnn.experimental.tensor.CoreCoord(0, 0),
+                ttnn.experimental.tensor.CoreCoord(ln_num_cores_x - 1, ln_num_cores_y - 1),
+            ),
+        }
+    )
+
+    model_config["LAYERNORM_BLOCK_SHARDED_MEM_CFG"] = ttnn.experimental.tensor.MemoryConfig(
+        ttnn.experimental.tensor.TensorMemoryLayout.BLOCK_SHARDED,
+        ttnn.experimental.tensor.BufferType.L1,
+        ttnn.experimental.tensor.ShardSpec(
+            core_range_block_sharded_layernorm,
+            [
+                ln_shard_height_hidden_dim,
+                ln_shard_width_hidden_dim,
+            ],
+            ttnn.experimental.tensor.ShardOrientation.ROW_MAJOR,
+            False,
+        ),
+    )
+
+    layernorm_block_sharded_prg_config_inplace = (
+        ttnn.experimental.operations.primary.LayerNormShardedMultiCoreProgramConfig(
+            compute_with_storage_grid_size=[ln_num_cores_x, ln_num_cores_y],
+            subblock_w=6,
```

Comment:
More of a question than a comment, how was this number determined?

Reply:
Valid range of subblock_w is [1-8], which is the number of tiles that dest register holds at one moment. So you essentially split block_w into subblock_w factors. block_w is 18 in this case, so the biggest factor less than 8 is 6. (It's poorly hardcoded here :D, will make a formula that calculates it)

Reply:
Ah ok, thanks :)